waiter/src/waiter/process_request.clj

```diff
@@ -242,14 +243,10 @@
       (try
         (loop [unreported-bytes-to-statsd 0]
           (let [available-bytes (.available input-stream)
-                ;; get a buffer size between buffer-increment-size and max-buffer-size
+                ;; get a buffer size between min-buffer-size and max-buffer-size
                 buffer-size (-> available-bytes
-                              (max 0)
-                              double
-                              (/ buffer-increment-size)
-                              int
-                              inc
-                              (* buffer-increment-size)
+                              (bit-and buffer-increment-mask)
```

Comment:
Nitpick: The indentation is off here, but it looks like it may have been off before your patch as well. The `(bit-and` line should be aligned with `available-bytes` above, rather than being indented 2 spaces like an `(if` block.